eng/pipelines/templates/steps/analyze.yml

```diff
@@ -4,18 +4,60 @@ parameters:
     default: stable
 
 steps:
+  - ${{ if ne(parameters.Toolchain, 'nightly') }}:
+      - template: /eng/pipelines/templates/steps/use-rust.yml@self
+        parameters:
+          Toolchain: nightly
+
```

Comment:
Why did you put this back after removing it from the other PR? We don't need to run this on every toolchain - just nightly where it'll work until stabilized. Even then, we only need to run it on, say, stable.

Reply:
We only run the Analyze job with stable, but we have to use nightly to run the rust script.  This bakes out to a single invocation of the "Run source analysis" step with toolchain == stable, but with the invocations:

```
/> cargo +stable check -p azure_core --no-default-features
/> cargo +stable fmt --all -- --check
/> cargo +stable clippy --all
/> cargo +stable doc --all --no-deps

/sdk/typespec> cargo +nightly -Zscript /mnt/vss/_work/1/s/eng/scripts/verify-dependencies.rs
/sdk/core> cargo +nightly -Zscript /mnt/vss/_work/1/s/eng/scripts/verify-dependencies.rs
...
```